pkg/kubelet/kubelet.go

```diff
@@ -1181,9 +1200,33 @@ func (kl *Kubelet) setupDataDirs() error {
 	if err := os.MkdirAll(kl.getPodsDir(), 0750); err != nil {
 		return fmt.Errorf("error creating pods directory: %v", err)
 	}
+	if err := kl.chownDirForRemappedIDs(kl.getPodsDir()); err != nil {
+		return fmt.Errorf("error chowning pods directory: %v", err)
+	}
 	if err := os.MkdirAll(kl.getPluginsDir(), 0750); err != nil {
 		return fmt.Errorf("error creating plugins directory: %v", err)
 	}
+	if err := kl.chownDirForRemappedIDs(kl.getPluginsDir()); err != nil {
+		return fmt.Errorf("error chowning plugins directory: %v", err)
+	}
+
+	return nil
+}
+
+// chownDirForRemappedIDs change dir and path ownerships if NodeUserNamespace is enabled by feature-gate
+func (kl *Kubelet) chownDirForRemappedIDs(path string) error {
+	if !kl.nodeWideUserNamespaceRemapping {
+		// No-Op is NodeUserNamespace feature gate is not enabled
+		return nil
+	}
+	if path == "" {
+		return fmt.Errorf("path to be setup is empty.")
+	}
+	uid, gid := kl.getRemappedIDs()
```

Comment:
what do these remapped ids represent? there's no pod/container-specific input here... what uid/gid are going to be global to the kubelet?

Reply:
these are non-zero ids on host which what mapped to uid 0 and gid 0 in container's user namespace. UID 0 in all containers on a node will be mapped to this non-zero uid on host.

Reply:
@vikaschoudhary16  when a user specifies runAsUser/runAsGroup, i am assuming the uid/gid mentioned is from the containers namespace and not the pods namespace right ?

Reply:
i guess you meant "containers namespace and not host namespace". Yes, runAsUser/runAsGroup mentioned in the will be from container namespace.
for example if runtime usernamepsace configuration is "testuser:80000:100" and uid 60 is provided in security context, in container it will be uid 60 but on host it will be 800060